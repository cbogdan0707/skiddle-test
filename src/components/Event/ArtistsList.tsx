import React from 'react';
import { Link } from 'react-router-dom';
import { ArtistsData, EventData } from '../../types/types';

export const ArtistsList = ({
  response,
}: {
  response: { results: EventData };
}) => {
  return (
    <section>
      {response.results.artists && response.results.artists.length !== 0 && (
        <div className="skiddle-u-color-primary skiddle-u-mb-medium skiddle-c-artist-title">
          List of artists:
        </div>
      )}
      <ul className="skiddle-c-search-results-container">
        {response.results.artists &&
          response.results.artists.map((artist: ArtistsData) => (
            <li key={artist.artistid}>
              <Link
                className="skiddle-u-primary-button skiddle-u-pa-small"
                to={{ pathname: `/artist/${artist.artistid}` }}
              >
                {artist.name}
              </Link>
            </li>
          ))}
      </ul>
    </section>
  );
};
