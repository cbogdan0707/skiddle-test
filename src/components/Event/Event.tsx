import React, { ReactElement } from 'react';
import { config } from '../../utils/config';
import { ArtistsList } from './ArtistsList';
import { useGetEventsOrArtists } from '../../utils/hooks';

export const Event = ({
  match,
}: {
  match: { params: { id: string } };
}): ReactElement => {
  const { hasError, response } = useGetEventsOrArtists(
    config.eventDetailsUrl,
    match,
    config.apiKey
  );

  return (
    <div>
      {hasError && (
        <div data-testid="fetch-error">Error: failed to fetch events</div>
      )}
      {!hasError && response.results && (
        <>
          <div className="skiddle-u-text-align-center skiddle-u-background-primary">
            <img
              className="skiddle-u-mt-large"
              alt="event logo"
              src={response.results.largeimageurl}
            />
            <h1 className="skiddle-u-color-white skiddle-u-pb-large">
              {response.results.eventname}
            </h1>
          </div>
          <ArtistsList response={response} />
        </>
      )}
    </div>
  );
};
