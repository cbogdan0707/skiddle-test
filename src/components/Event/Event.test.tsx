import React from 'react';
import { render } from '@testing-library/react';
import { Event } from './Event';

describe('Event', () => {
  it('should match snapshot', () => {
    const { asFragment } = render(
      <Event match={{ params: { id: '2333333' } }} />
    );
    expect(asFragment()).toMatchSnapshot();
  });
});
