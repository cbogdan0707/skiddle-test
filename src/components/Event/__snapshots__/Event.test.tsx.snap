// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Event should match snapshot 1`] = `
<DocumentFragment>
  <div>
    <div
      class="skiddle-u-text-align-center skiddle-u-background-primary"
    >
      <img
        alt="event logo"
        class="skiddle-u-mt-large"
      />
      <h1
        class="skiddle-u-color-white skiddle-u-pb-large"
      />
    </div>
    <section>
      <ul
        class="skiddle-c-search-results-container"
      />
    </section>
  </div>
</DocumentFragment>
`;
